jlm/llvm/backend/rvsdg2jlm/rvsdg2jlm.cpp

```diff
@@ -515,21 +516,20 @@ convert_node(const rvsdg::Node & node, context & ctx)
 {
   static std::
       unordered_map<std::type_index, std::function<void(const rvsdg::Node & node, context & ctx)>>
-          map({ { typeid(lambda::operation), convert_lambda_node },
-                { std::type_index(typeid(rvsdg::GammaOperation)), convert_gamma_node },
-                { std::type_index(typeid(rvsdg::ThetaOperation)), convert_theta_node },
-                { typeid(phi::operation), convert_phi_node },
-                { typeid(delta::operation), convert_delta_node } });
+          map({ { typeid(rvsdg::LambdaNode), convert_lambda_node },
+                { std::type_index(typeid(rvsdg::GammaNode)), convert_gamma_node },
+                { std::type_index(typeid(rvsdg::ThetaNode)), convert_theta_node },
+                { typeid(phi::node), convert_phi_node },
+                { typeid(delta::node), convert_delta_node } });
 
-  if (dynamic_cast<const rvsdg::SimpleOperation *>(&node.GetOperation()))
+  if (dynamic_cast<const rvsdg::SimpleNode *>(&node))
```

Comment:
Since you touch this line anyway: `is<>`

Reply:
No this does not work, is<> checks for the operation while I want code to really rather check for node type primarily.

Generalizing is<> to accept either might be possible but requires interesting C++ "concepts" which I would rather not introduce.

I would indeed generally like to remove the duality of dispatching on "op" vs. "node", and cleanly settle that "op" is only sensible for things that are SimpleNodes (you still need to cast the node to something specific for doing anything sensible with structural nodes, so there is no point in ever starting with op there).